src/daemon/daemon.cpp

```diff
@@ -1972,13 +1972,15 @@ grpc::Status mp::Daemon::shutdown_vm(VirtualMachine& vm, const std::chrono::mill
     {
         delayed_shutdown_instances.erase(name);
 
-        mp::optional<mp::SSHSession> session{mp::nullopt};
+        mp::optional<mp::SSHSession> session;
         try
         {
             session = mp::SSHSession{vm.ssh_hostname(), vm.ssh_port(), vm.ssh_username(), *config->ssh_key_provider};
         }
-        catch (...)
+        catch (const std::exception& e)
```

Comment:
For the record, as [previously discussed](https://github.com/CanonicalLtd/multipass/pull/589#discussion_r253998933), I am not keen on this, but don't want to insist.